server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java

```diff
@@ -754,4 +755,127 @@ public void onRejection(Exception e) {
         executor.execute(shouldBeRejected);
         assertTrue(rejected.get());
     }
+
+    public void testScalingWithEmptyCore() {
+        testScalingWithEmptyCore(
+            EsExecutors.newScaling(
+                getTestName(),
+                0,
+                1,
+                0,
+                TimeUnit.MILLISECONDS,
+                true,
+                EsExecutors.daemonThreadFactory(getTestName()),
+                threadContext
+            )
+        );
+    }
+
+    public void testScalingWithEmptyCoreAndKeepAlive() {
+        testScalingWithEmptyCore(
+            EsExecutors.newScaling(
+                getTestName(),
+                0,
+                1,
+                1,
+                TimeUnit.MILLISECONDS,
+                true,
+                EsExecutors.daemonThreadFactory(getTestName()),
+                threadContext
+            )
+        );
+    }
+
+    public void testScalingWithEmptyCoreAndWorkerPoolProbing() {
+        // https://github.com/elastic/elasticsearch/issues/124667 is difficult to reproduce if max pool size > 1.
```

Comment:
Difficult, but probably not impossible. I would be happier to see tests for both the `=1` and `>1` cases.

Reply:
I'll follow up on this and will give this another try next week 👍